### DISCUSSION TITLE
Getting function parameters from LLIL_CALL / LLIL_SYSCALL

### DISCUSSION THREAD

**Discussion by crand6** [Vector35/binaryninja-api#2228] [Analysis] [answered]
I notice that `LLIL_CALL` only has a `dest` operand, while the `LLIL_CALL_SSA` includes addition info such as `output`, `stack`, and `param`. This extra information seems to me like it should have come from the calling convention. To help my understanding:

1) Why is this information tied into the SSA form? I originally anticipated I would get information like that from one of the `bv.arch.calling_conventions`.
2) Is just getting the `LowLevelILInstruction.ssa_form` sufficient to then get what registers/stack addresses would be used to pass arguments? Or would I be missing additional information by not doing an analysis on an entire LowLevelILFunction? (In my case, I just want to know what registers are used generally (the calling convention), not the actual possible value of an SSA variable).
3) What are the pros/cons of providing this information through SSA form instead of something like a "calling convention" operand?

**Comment by plafosse** [comment, marked as answer]
That information is generated from SSA, its doesn't exist in LLIL (non-ssa) because we haven't generated that information yet. It sounds like MLIL will be a better bet for doing the type of analysis that you're trying to do.
